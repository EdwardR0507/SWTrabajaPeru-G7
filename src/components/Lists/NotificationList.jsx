import React from "react";
import ClickAwayListener from "@material-ui/core/ClickAwayListener";
import Grow from "@material-ui/core/Grow";
import Paper from "@material-ui/core/Paper";
import Popper from "@material-ui/core/Popper";
import MenuItem from "@material-ui/core/MenuItem";
import MenuList from "@material-ui/core/MenuList";
import { makeStyles } from "@material-ui/core/styles";
import NotificationsIcon from "@material-ui/icons/Notifications";
const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
  },
  paper: {
    marginRight: theme.spacing(2),
  },
}));

export default function MenuListComposition() {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);
  const anchorRef = React.useRef(null);

  const handleToggle = () => {
    setOpen(!open);
  };

  const handleClose = (event) => {
    if (anchorRef.current && anchorRef.current.contains(event.target)) {
      return;
    }

    setOpen(false);
  };

  function handleListKeyDown(event) {
    if (event.key === "Tab") {
      event.preventDefault();
      setOpen(false);
    }
  }

  // return focus to the button when we transitioned from !open -> open
  const prevOpen = React.useRef(open);
  React.useEffect(() => {
    if (prevOpen.current === true && open === false) {
      anchorRef.current.focus();
    }

    prevOpen.current = open;
  }, [open]);

  return (
    <div className={classes.root}>
      <div>
        <NotificationsIcon
          fontSize="large"
          style={{ color: "#FFF" }}
          role="toggle"
          ref={anchorRef}
          aria-controls={open ? "menu-list-grow" : undefined}
          aria-haspopup="true"
          onClick={handleToggle}
        />

        <Popper
          open={open}
          anchorEl={anchorRef.current}
          role={undefined}
          transition
          disablePortal
        >
          {({ TransitionProps, placement }) => (
            <Grow
              {...TransitionProps}
              style={{
                transformOrigin:
                  placement === "bottom" ? "center top" : "center bottom",
              }}
            >
              <Paper>
                <ClickAwayListener role="close" onClickAway={handleClose}>
                  <MenuList
                    autoFocusItem={open}
                    id="menu-list-grow"
                    onKeyDown={handleListKeyDown}
                  >
                    <MenuItem onClick={handleClose}>Profile</MenuItem>
                    <MenuItem onClick={handleClose}>My account</MenuItem>
                    <MenuItem onClick={handleClose}>Logout</MenuItem>
                  </MenuList>
                </ClickAwayListener>
              </Paper>
            </Grow>
          )}
        </Popper>
      </div>
    </div>
  );
}
