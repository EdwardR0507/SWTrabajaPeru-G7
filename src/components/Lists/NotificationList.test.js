import React from "react";
import '@testing-library/jest-dom/extend-expect';
import { shallow } from "enzyme";
import NotificationList from "./NotificationList";

describe("<NotificationList />", () => {
    it('render "NotificationList"', () => {
        const wrapper = shallow(<NotificationList />);
        expect(wrapper).toMatchSnapshot()
      }); 
      it("test handleToogle", () => {
        const wrapper = shallow(<NotificationList />);
        const button = wrapper.find({role: "toggle"})
        const open = wrapper.find({role: "toggle"})
        button.simulate("click")
        expect(open.props("aria-controls")).toBe(true)
      })
      it("test handleClose", () => {
        const wrapper = shallow(<NotificationList />);
        const button = wrapper.find({role: "close"})
        const open = wrapper.find({role: "toggle"})
        button.simulate("click")
        expect(open.props("aria-controls")).toBe(false)
      })
}) 