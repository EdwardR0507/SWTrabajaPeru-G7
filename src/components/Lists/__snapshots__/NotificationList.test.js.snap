// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationList /> render "NotificationList" 1`] = `ShallowWrapper {}`;
